### PR TITLE
gitleaks: 8.20.1 -> 8.21.0

### PR DESCRIPTION
**Pull request by 9999years**
Also add `nix-update-script`.

https://github.com/gitleaks/gitleaks/releases/tag/v8.21.0

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/tools/security/gitleaks/default.nix [hunk 1]
```diff
@@ -44,6 +45,8 @@ buildGoModule rec {
     command = "${pname} version";
   };
 
+  passthru.updateScript = nix-update-script { };
```
Comment (fabaff) [inline pkgs/tools/security/gitleaks/default.nix:48]: Bot and `nix-update` worked fine in the past. Does adding `nix-update-script` solve an issue or is it just for convenience?

Comment (9999years) [reply pkgs/tools/security/gitleaks/default.nix:48]: It let me run `nix-shell maintainers/scripts/update.nix --argstr package gitleaks` to generate this PR, which did not work before adding this.

This is the current advice in `pkgs/README.md`; if the situation has changed, please open a PR for that file: https://github.com/NixOS/nixpkgs/blob/7e9f7ce40331de889033e2e82b2e1cd96f7cdabe/pkgs/README.md#automatic-package-updates

Comment (fabaff) [reply pkgs/tools/security/gitleaks/default.nix:48]: Manually updating gitleaks with `nix-update` adds the changelog and the diff to the commit message which also forms the content for the PR description. 

https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#writing-good-commit-messages
> Package version upgrades usually allow for simpler commit messages, including attribute name, old and new version, as well as a reference to the relevant release notes/changelog.

`nix-shell maintainers/scripts/update.nix --argstr package gitleaks` doesn't do [this]. I highly doubt that people will start using `update.nix` over `nix-update` when updating packages manually, if not necessary, as you have to create the commit message by yourself.

The README (https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#automatic-package-updates) also says:

> While not strictly necessary if the project is listed on [Repology](https://repology.org/)...

`passthru.updateScript = nix-update-script { };` would then only be for convenience.

Comment (9999years) [reply pkgs/tools/security/gitleaks/default.nix:48]: > Manually updating gitleaks with `nix-update` adds the changelog and the diff to the commit message which also forms the content for the PR description.

Thanks, I didn't know this! This is the first I'm hearing that we should run `nix-update` directly. Could you point me to the documentation that explains how to use it? All I see [in the README](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#how-are-update-scripts-executed) is this note, which explicitly says to run `update.nix`:

> Update scripts are to be invoked by the [automatic package update script](https://github.com/NixOS/nixpkgs/blob/master/maintainers/scripts/update.nix). You can run `nix-shell maintainers/scripts/update.nix` in the root of Nixpkgs repository for information on how to use it.

`nix-update` is mentioned in passing ("A common pattern is to use the [`nix-update-script`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/ni/nix-update/nix-update-script.nix) attribute provided in Nixpkgs, which runs [`nix-update`](https://github.com/Mic92/nix-update)"), but it doesn't explain how to run `nix-update` yourself or recommend doing so.

I'm happy to run `nix-update` directly if that's the new recommendation, but I can't find any documentation to corroborate this.

Comment (9999years) [reply pkgs/tools/security/gitleaks/default.nix:48]: I'm fine with this line being here purely for convenience. I encourage you to open a PR updating the `pkgs/README.md` documentation if you'd like to steer people towards `nix-update`.

**Review by lf- — APPROVED** [review]
thank you so much for pushing back on adhoc undocumented knowledge. we need to better document the way to do updates for sure.

**Pull request merged**